src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +600,275 @@ class RigidBodyComponentSystem extends ComponentSystem {
         return results;
     }
 
+    /**
+     * Perform a collision check on the world and return all entities the box hits.
+     * It returns an array of {@link HitResult}. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {Vec3} halfExtents - The half-extents of the box in the x, y and z axes. Cast distance will be added to Z-Axis.
+     * @param {Vec3} start - The world space point where the box starts.
+     * @param {Vec3} end - The world space point where the box ends.
+     *
+     * @returns {HitResult[]} An array of boxCast hit results (0 length if there were no hits).
+     */
+    boxCast(halfExtents, start, end) {
+        // Sweeping
+        halfExtents.z += start.distance(end) / 2;
+
+        // Find rotation
+        shapecastRotationMatrix.setLookAt(start, end, Vec3.UP);
+        shapecastRotation.setFromMat4(shapecastRotationMatrix);
+
+        // Transform start vector to make it beween initial start and end.
+        start.add(end).divScalar(2);
+
+        return this._shapeTest(new Ammo.btBoxShape(halfExtents), start, shapecastRotation);
```

Comment:
How does this work, don't you need to use `ammoVec3` instead of `halfExtents` here (`pc.Vec3` vs `Ammo.btVector3`) to create a `btBoxShape`?

Reply:
Oh yeah, my bad, had it locally but not on commit, that's updated!